lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -56,8 +60,53 @@ pub fn condition_converter<'a>(
                 .collect();
             Box::new(move |point_id| segment_ids.contains(&point_id))
         }
+        Condition::Nested(nested) => {
+            // Select indexes for nested fields. Trim nested part from key, so
+            // that nested condition can address fields without nested part.
+
+            // Example:
+            // Index for field `nested.field` will be stored under key `nested.field`
+            // And we have a query:
+            // {
+            //   "nested": {
+            //     "path": "nested",
+            //     "filter": {
+            //         ...
+            //         "match": {"key": "field", "value": "value"}
+            //     }
+            //   }
+
+            // In this case we want to use `nested.field`, but we only have `field` in query.
+            // Therefore we need to trim `nested` part from key. So that query executor
+            // can address proper index for nested field.
+            let nested_path = nested.array_key();
+
+            let nested_indexes = select_nested_indexes(&nested_path, field_indexes);
+
+            Box::new(move |point_id| {
+                payload_provider.with_payload(point_id, |payload| {
+                    let field_values = payload.get_value(&nested_path).values();
+
+                    for value in field_values {
+                        if let Value::Object(object) = value {
+                            let get_payload = || OwnedPayloadRef::from(object);
+                            if check_payload(
+                                get_payload,
+                                |_| None,
```

Comment:
we need a comment here explaining why the closure is harcoded to return None

Reply:
because has_id in nested is not supported. So it just always return none and check is always false

Reply:
Sorry for the misunderstanding, I meant comment in the code :)